src/app/components/client/PlusShutdownBanner.tsx

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import styles from "./PlusShutdownBanner.module.scss";
+import { Button } from "./Button";
+import { CloseBtn } from "../server/Icons";
+import { useL10n } from "../../hooks/l10n";
+import { useLocalDismissal } from "../../hooks/useLocalDismissal";
+import { useHasRenderedClientSide } from "../../hooks/useHasRenderedClientSide";
+import { type ShutdownState } from "../../functions/server/getPlusShutdownState";
+import { FeatureFlagName } from "../../../db/tables/featureFlags";
+
+type Props = {
+  shutdownState: ShutdownState;
+  countryCode: string;
+  enabledFeatureFlags: FeatureFlagName[];
+};
+
+export const PlusShutdownBanner = (props: Props) => {
+  const l10n = useL10n();
+
+  const currentMoment =
```

Comment:
personally I think it would be simpler if this banner type logic was moved to `getShutdownState`. So `getShutdownState` would return the banner type e.g.  `"runup" | "shutdown" | "none" // or null`, given the current date and configuration. Then it's easy to unit-test the core banner choice logic. `isPremium` could be passed into the Props instead